src/storage/raw/raw_mvcc.rs

```diff
@@ -120,60 +120,64 @@ impl<I: Iterator> RawMvccIterator<I> {
         self.is_valid = None;
     }
 
-    fn move_to_prev_max_ts(&mut self, res: Result<bool>) -> Result<bool> {
-        if *res.as_ref().unwrap() && self.inner.valid()? {
+    fn move_to_prev_max_ts(&mut self) -> Result<bool> {
+        if self.inner.valid()? {
             self.update_cur_kv(self.inner.key().to_vec(), self.inner.value().to_vec());
             self.inner.prev()?;
         } else {
             self.clear_cur_kv();
             return Ok(false);
         }
-        while *res.as_ref().unwrap() && self.inner.valid()? {
+        while self.inner.valid()? {
+            // cur_key should not be None here.
             if is_user_key_eq(self.cur_key.as_ref().unwrap(), self.inner.key()) {
                 self.update_cur_kv(self.inner.key().to_vec(), self.inner.value().to_vec());
                 self.inner.prev()?;
             } else {
                 break;
             }
         }
-        res
+        Ok(true)
     }
 }
 
 impl<I: Iterator> Iterator for RawMvccIterator<I> {
     fn next(&mut self) -> Result<bool> {
         let cur_key = self.inner.key().to_owned();
-        let mut res = self.inner.next();
-        while *res.as_ref().unwrap()
-            && self.inner.valid()?
-            && is_user_key_eq(&cur_key, self.inner.key())
-        {
-            res = self.inner.next();
+        let mut res = self.inner.next()?;
```

Comment:
I think it's better to make sure why it's chosen to panicked in the old code.

Reply:
actually this file is newly added in this pr. Panic is a bug, i think, as it's a common case that seek_for_prev return error when disk error or something else. In this case, we should return the error to outside other than panic.

Reply:
thanks for pointing this issue.